src/strategyHelpers/blueGreen/route.ts

```diff
@@ -0,0 +1,146 @@
+import {Kubectl} from '../../types/kubectl'
+import {sleep} from '../../utilities/timeUtils'
+import {RouteStrategy} from '../../types/routeStrategy'
+import {
+   BlueGreenDeployment,
+   BlueGreenManifests,
+   getManifestObjects,
+   GREEN_LABEL_VALUE,
+   deployObjects
+} from './blueGreenHelper'
+
+import {
+   getUpdatedBlueGreenIngress,
+   isIngressRouted
+} from './ingressBlueGreenHelper'
+import {getUpdatedBlueGreenService} from './serviceBlueGreenHelper'
+import {createTrafficSplitObject} from './smiBlueGreenHelper'
+
+import * as core from '@actions/core'
+import {TrafficSplitObject} from '../../types/k8sObject'
+import {getBufferTime} from '../../inputUtils'
+
+export async function routeBlueGreenForDeploy(
+   kubectl: Kubectl,
+   inputManifestFiles: string[],
+   routeStrategy: RouteStrategy
+): Promise<BlueGreenDeployment> {
+   // sleep for buffer time
+   const bufferTime: number = getBufferTime()
+   if (bufferTime < 0 || bufferTime > 300)
+      throw Error('Version switch buffer must be between 0 and 300 (inclusive)')
```

Comment:
you can move this if statement to inside the input.ts. Would make more sense there since inputs are handled there